### PR TITLE
[Go] client-side-validation flag support

### PR DESCRIPTION
**Pull request by mcardosos**
PTAL @marstr @jhendrixMSFT @olydis 
Fixes https://github.com/Azure/autorest/issues/2100 for Go generator

### CODE REVIEW

File: src/generator/AutoRest.Go/GeneratorSettingsGo.cs [hunk 1]
```diff
@@ -12,5 +12,10 @@ public class GeneratorSettingsGo : IsSingleton<GeneratorSettingsGo>, IGeneratorS
         public virtual string Name => "Go";
 
         public virtual string Description => "Generic Go code generator.";
+
+        /// <summary>
+        /// Indicates whether to generate client side validation.
+        /// </summary>
+        public bool ClientSideValidation { get; set; }
```
Comment (jhendrixMSFT) [inline src/generator/AutoRest.Go/GeneratorSettingsGo.cs]: Where is this set?

File: src/generator/AutoRest.Go/Model/MethodGo.cs [hunk 2]
```diff
@@ -213,6 +214,10 @@ public string MethodReturnSignature(bool helper)
             return $"{retVal}, {errVal}";
         }
 
+        public bool ShouldValidate() {
+            return !ParameterValidations.IsNullOrEmpty();
```
Comment (jhendrixMSFT) [inline src/generator/AutoRest.Go/Model/MethodGo.cs]: Nit, opening curly brace is in the wrong location.  More importantly shouldn't this return Singleton<GeneratorSettingsGo>.Instance.ClientSideValidation?  Then you can omit the changes in ParameterGo.cs.

Comment (mcardosos) [inline src/generator/AutoRest.Go/Model/MethodGo.cs]: I like the changes in ParameterGo.cs. Why gen code that wont be written to the file anyways. Still will change code here.

---
In reply to: [130652897](https://github.com/Azure/autorest/pull/2490#discussion_r130652897) [](ancestors = 130652897)

File: src/generator/AutoRest.Go/GeneratorSettingsGo.cs (hunk 1 above)
Comment (mcardosos) [inline src/generator/AutoRest.Go/GeneratorSettingsGo.cs]: AutoRest core

File: src/generator/AutoRest.Go/Model/MethodGo.cs (hunk 2 above)
Comment (mcardosos) [inline src/generator/AutoRest.Go/Model/MethodGo.cs]: Fixed

Comment (jhendrixMSFT) [inline src/generator/AutoRest.Go/Model/MethodGo.cs]: But will ParameterGo.Validate() actually get called when ShouldValidate() == false?  I would think it not to get called.

**Comment by olydis** [comment]
moved to https://github.com/Azure/autorest.go/pull/6

**Pull request closed without merging**